### DISCUSSION TITLE
Adding tooltips to data points

### DISCUSSION THREAD

**Discussion by simonalexander2005** [GSI-CS-CO/chart-fx#495] [Q&A] [answered]
Hi,

I would like to add tooltips to individual data points. In raw JFX Charts, I've been doing it as follows:

```
for (XYChart.Series<Number, Number> s : scatterChartTlk.getData()) 
{
    for (XYChart.Data<Number, Number> d : s.getData()) 
    {
        Tooltip t =  new Tooltip(epochSecondsToStringConverter.toString(d.getXValue()) + ", took " + d.getYValue().intValue() + " ms");
	Tooltip.install(d.getNode(), t);
    }
}
```

What would be the equivalent of this in `chart-fx`?

**Comment by wirew0rm** [comment, marked as answer]
Hey,

take a look at our samples, for example the `MultipleAxesSample` which shows how to use the `DataPointTooltip` plugin. Your approach does not work because chartfx doesn't instantiate a node for each data point for performance reasons but instead uses the Graphics context directly. If you need to customise the tooltip, you can do so by overriding the default Implementation.

Hope this helps,

Alex
